tests/collections/tts/models/test_fastpitch.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This file implemented unit tests for loading all pretrained FastPitch NGC checkpoints and generating Mel-spectrograms.
+The test duration breakdowns are shown below. In general, each test for a single model is ~25 seconds on an NVIDIA RTX A6000.
+"""
+import pytest
+
+from nemo.collections.tts.models import FastPitchModel
+
+available_models = [model.pretrained_model_name for model in FastPitchModel.list_available_models()]
+
+
+@pytest.fixture(params=available_models, ids=available_models)
+@pytest.mark.run_only_on('GPU')
+def pretrained_model(request):
+    model_name = request.param
+    model = FastPitchModel.from_pretrained(model_name=model_name)
+    return model, model_name
+
+
+@pytest.mark.nightly
+@pytest.mark.run_only_on('GPU')
+def test_inference(pretrained_model, language_specific_text_example):
+    model, model_name = pretrained_model
```

Comment:
When I add a model I am always worried that the model name and URL are incorrectly formatted. Would it be worthwhile to add a check that validates that the URL matches a regex for the correct format?

Like https://api.ngc.nvidia.com/v2/models/nvidia/nemo/tts_de_fastpitchhifigan/versions/1.10.0/files/tts_de_fastpitch_align.nemo

But replace the version number with any digit and "tts_de_fastpitchhifigan" with the model_name.

Reply:
agreed. But it seems there are mismatches between `PretrainedModelInfo.pretrained_model_name` and the basename of `PretrainedModelInfo.location` in our current `list_available_models`. Let's discuss a naming convention for those names, and add a strict check in a separate PR.